stock_picking_invoice_link/tests/test_stock_picking_invoice_link.py

```diff
@@ -0,0 +1,134 @@
+class TestStockPickingInvoiceLink(TestSale):
+
+    def _update_product_qty(self, product, location):
+        product_qty = self.env['stock.change.product.qty'].create({
+            'location_id': location.id,
+            'product_id': product.id,
+            'new_quantity': 100.0,
+        })
+        product_qty.change_product_qty()
+        return product_qty
+
+    def setUp(self):
+        super(TestStockPickingInvoiceLink, self).setUp()
+        company = self.env.ref('base.main_company')
+        warehouse = self.env['stock.warehouse'].search(
+            [('company_id', '=', company.id)], limit=1)
+        stock_location = warehouse.lot_stock_id
+        for (_, p) in self.products.items():
+            if p.type == 'product':
+                self._update_product_qty(p, stock_location)
+        prod_order = self.products['prod_order']
+        prod_del = self.products['prod_del']
+        serv_order = self.products['serv_order']
+        self.so = self.env['sale.order'].create({
+            'partner_id': self.partner.id,
+            'partner_invoice_id': self.partner.id,
+            'partner_shipping_id': self.partner.id,
+            'order_line': [
+                (0, 0, {
+                    'name': prod_order.name, 'product_id': prod_order.id,
+                    'product_uom_qty': 2, 'product_uom': prod_order.uom_id.id,
+                    'price_unit': prod_order.list_price
+                }),
+                (0, 0, {
+                    'name': prod_del.name, 'product_id': prod_del.id,
+                    'product_uom_qty': 2, 'product_uom': prod_del.uom_id.id,
+                    'price_unit': prod_del.list_price
+                }),
+                (0, 0, {
+                    'name': serv_order.name, 'product_id': serv_order.id,
+                    'product_uom_qty': 2, 'product_uom': serv_order.uom_id.id,
+                    'price_unit': serv_order.list_price
+                }),
+            ],
+            'pricelist_id': self.env.ref('product.list0').id,
+            'picking_policy': 'direct',
+        })
+        self.so.action_confirm()
+
+    def test_00_sale_stock_invoice_link(self):
+        inv_obj = self.env['account.invoice']
+        self.assertTrue(self.so.picking_ids,
+                        'Sale Stock: no picking created for '
+                        '"invoice on delivery" stockable products')
+
+        # invoice on order
+        self.so.action_invoice_create()
+
+        # deliver partially
+        self.assertEqual(self.so.invoice_status, 'no',
+                         'Sale Stock: so invoice_status should be '
+                         '"nothing to invoice" after invoicing')
+        pick_1 = self.so.picking_ids.filtered(
+            lambda x: x.picking_type_code == 'outgoing' and
+            x.state in ('confirmed', 'assigned', 'partially_available'))
+        pick_1.force_assign()
+        pick_1.move_line_ids.write({'qty_done': 1})
+        wiz_act = pick_1.do_new_transfer()
```

Comment:
You have to check how this is handled in v11 and change the test accordingly

Reply:
Yup, but @celm1990 made another PR... Should I continue with this?